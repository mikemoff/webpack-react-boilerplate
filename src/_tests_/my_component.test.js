import MyComponent from '../js/components/my_component';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';


describe('MyComponent', () => {
  
  it('should be defined', () => {
    expect(MyComponent).toBeDefined();
  });

  it('should render a title', () => {
    const custom = 'Component Title';
    const comp = shallow(<MyComponent title={custom} />);
    expect(comp).toMatchSnapshot();
  });


});
