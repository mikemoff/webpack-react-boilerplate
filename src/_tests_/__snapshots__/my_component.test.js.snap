// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render a title 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyComponent
    title="Component Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Component Title
      </h1>,
      "className": "my-component",
      "id": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Component Title",
      },
      "ref": null,
      "rendered": "Component Title",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Component Title
        </h1>,
        "className": "my-component",
        "id": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Component Title",
        },
        "ref": null,
        "rendered": "Component Title",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
